nats.go

```diff
@@ -5455,6 +5456,66 @@ func (nc *Conn) GetClientID() (uint64, error) {
 	return nc.info.CID, nil
 }
 
+// ConnStatusChanged returns a channel on which connection status changes will be reported.
+// Reported statuses: CONNECTED, RECONNECTING, DISCONNECTED, CLOSED.
+func (nc *Conn) ConnStatusChanged() chan Status {
+	statuses := []Status{CONNECTED, RECONNECTING, DISCONNECTED, CLOSED}
+	ch := make(chan Status)
+	for _, s := range statuses {
+		nc.RegisterStatusChangeListener(s, ch)
+	}
+	return ch
+}
+
+// RegisterStatusChangeListener registers a channel waiting for a specific status change event.
+// Status change events are non-blocking - if no receiver is waiting for the status change,
+// it will not be sent on the channel. Closed channels are ignored.
+func (nc *Conn) RegisterStatusChangeListener(status Status, ch chan Status) {
```

Comment:
Maybe `RegisterStatusChangeListener(ch chan Status, status ...Status)` so one can easily listen for just a few statusses?

It's a bit weird being the only channel based event listener in this client, but seems like a nice addition and good use of the channel anyway

Reply:
I like the variadic idea, I was thinking maybe can make `ConnStatusChanged(...Status)` as a variadic with the defaults being `[]Status{CONNECTED, RECONNECTING, DISCONNECTED, CLOSED}` if nothing passed and keep only one (maybe `StatusChanges(...Status)` or `StatusChanged(...Status)`? so that it pairs better with the `Status() Status` API that is already there.